### ISSUE TITLE
Are those headers actually required?

### ISSUE THREAD

**Issue by brandonros** [Ximaz/instagram-downloader#3]
From the README:

> The required headers are the thing Instagram looks at to tell if a request has to be server or not. The most important ones are X-Mid, X-CSRFToken, ig_did's cookie, X-IG-App-ID and X-ASBD-ID.

```javascript
import fetch from 'node-fetch'
import { URLSearchParams } from 'url'

const getInstagramPosts = async ({ cookies, queryId, userId, numPosts, cursor }) => {
  const query = new URLSearchParams({
    query_hash: queryId,
    variables: JSON.stringify({
      id: userId,
      first: numPosts,
      after: cursor
    })
  })
  const response = await fetch(`https://www.instagram.com/graphql/query/?${query.toString()}`, {
    headers: {
      'cookie': cookies
    }
  })
  if (response.status !== 200) {
    throw new Error(`Invalid response status: ${response.status}`)
  }
  const responseBody = await response.json()
  return responseBody
}
```

Works for me without any `x-...` headers

**Comment by brandonros** [comment]
`ig_did`, `ds_user_id`, and `sessionid` cookies are required. All others seem to be optional.

**Comment by KoH-H** [comment]
I have a question. How to analyze the path of ConsumerLibCommons.js.

**Comment by KoH-H** [comment]
I have a question. How to analyze the path of ConsumerLibCommons.js. I didn't find that url in the source code of the page.

**Comment by Ximaz** [comment]
> `ig_did`, `ds_user_id`, and `sessionid` cookies are required. All others seem to be optional.

It should work without those headers, yeah, but my goal here was to try to get as much as header as possible so that I can try to fake a real guest account for a little bit longer, but I think it makes no difference, proxies may be a better idea, evne though if I remember correctly, Instagram had try to ban them as much as possible.

**Comment by Ximaz** [comment]
> I have a question. How to analyze the path of ConsumerLibCommons.js. I didn't find that url in the source code of the page.

I'm not sure to understand your question, may you open a new issue and ask again ?
